src/types.rs

```diff
@@ -1,11 +1,42 @@
-#[derive(Clone, Debug, Eq, Hash, PartialEq)]
+use std::fmt::{Debug, Error, Formatter};
+
+#[derive(Clone, Eq, Hash, PartialEq)]
 pub struct Bytes(pub Vec<u8>);
 
-impl From<Vec<u8>> for Bytes {
-    fn from(t: Vec<u8>) -> Self {
-        Bytes(t)
+impl<T> From<T> for Bytes
```

Comment:
Actually, could you remove this conversion to generic for now for this PR? a group of us talked about it in a pairing session going through some PR feedback before and right now I'm more comfortable keeping it as the concrete `Vec<u8>` rather than generic `T` here so that it's clearer to me when allocations happen. I expect to make it generic eventually but want to defer that for a bit.

Reply:
Yeah, I briefly thought about allocations and agree that it's not clear when it happens here.